testing/helpers.py

```diff
@@ -0,0 +1,54 @@
+from typing import Generator, Iterable, List, Optional, Tuple
+
+from flake8_annotations import Function, FunctionVisitor, PY_GTE_38
+from flake8_annotations.checker import TypeHintChecker
+from flake8_annotations.error_codes import Error
+
+if PY_GTE_38:
+    import ast
+else:
+    from typed_ast import ast3 as ast
+
+
+def parse_source(src: str) -> Tuple[ast.Module, List[str]]:
+    """Parse the provided Python source string and return an (typed AST, source) tuple."""
+    if PY_GTE_38:
+        # Built-in ast requires a flag to parse type comments
+        tree = ast.parse(src, type_comments=True)
+    else:
+        # typed-ast will implicitly parse type comments
+        tree = ast.parse(src)
+
+    lines = src.splitlines()
+
+    return tree, lines
+
+
+def check_source(src: str) -> Generator[Error, None, None]:
+    """Helper for generating linting errors from the provided source code."""
+    # Because TypeHintChecker is expecting a filename to initialize, rather than change this logic
+    # we can use this file as a dummy, then update its tree & lines attributes as parsed from source
+    checker_instance = TypeHintChecker(None, __file__)
+    checker_instance.tree, checker_instance.lines = parse_source(src)
+
+    return checker_instance.run()
+
+
+def functions_from_source(src: str) -> List[Function]:
+    """Helper for obtaining a list of Function objects from the provided source code."""
+    tree, lines = parse_source(src)
+    visitor = FunctionVisitor(lines)
+    visitor.visit(tree)
+
+    return visitor.function_definitions
+
+
+def find_matching_function(func_list: Iterable[Function], match_name: str) -> Optional[Function]:
+    """
+    Iterate over a list of Function objects & find the matching named function.
+
+    If no function is found, this returns None
+    """
+    for function in func_list:
```

Comment:
No issue, just to make sure you've seen it. Coverage reports that there's never a test case where `func_list` is empty, so the branch `52 -> exit` never runs during testing. I don't think an empty fun_list is a realistic case and it would just return `None` anyway (just as this function would if no matching function were to be found).

Reply:
I'm ok with not getting 100% coverage in the test suite as long as it's justifiable. In this case I haven't come up with a rewrite of the logic here that's worth changing to for the sake of getting to 100%; the function is pretty plain.

Reply:
That was my idea as well.